nifi-provenance-reporting-tasks/src/main/java/com/joeyfrazee/nifi/reporting/ElasticsearchProvenanceReporter.java

```diff
@@ -121,14 +116,20 @@ public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
     }
 
     public void indexEvent(final Map<String, Object> event, final ReportingContext context) throws IOException {
-        final String elasticsearchUrl = context.getProperty(ELASTICSEARCH_URL).getValue();
+        // Get properties from context.
+        final URL elasticsearchUrl = new URL(context.getProperty(ELASTICSEARCH_URL).getValue());
         final String elasticsearchIndex = context.getProperty(ELASTICSEARCH_INDEX).evaluateAttributeExpressions()
                 .getValue();
         final String elasticsearchCACertFingerprint = context.getProperty(ELASTICSEARCH_CA_CERT_FINGERPRINT).getValue();
         final String elasticsearchUser = context.getProperty(ELASTICSEARCH_USER).getValue();
         final String elasticsearchPassword = context.getProperty(ELASTICSEARCH_PASSWORD).getValue();
-        final ElasticsearchClient client = getElasticsearchClient(new URL(elasticsearchUrl),
-                elasticsearchCACertFingerprint, elasticsearchUser, elasticsearchPassword);
+
+        // Create the Elasticsearch API client.
+        final String protocol = elasticsearchUrl.getProtocol();
+        final RestClient restClient = (protocol.equals("https")) ? getSecureRestClient(elasticsearchUrl, elasticsearchCACertFingerprint, elasticsearchUser, elasticsearchPassword) : getRestClient(elasticsearchUrl);
```

Comment:
Note, the brackets around the statement are not necessary.

```
final RestClient restClient = protocol.equals("https")
   ? getSecureRestClient(elasticsearchUrl, elasticsearchCACertFingerprint, elasticsearchUser, elasticsearchPassword)
   :getRestClient(elasticsearchUrl);
```

Cleanup at some stage, not critical, but it unncessary noise.